service/db.go

```diff
@@ -1,92 +1,128 @@
 package service
 
 import (
+	"database/sql"
 	"fmt"
+	pb "github.com/hwsc-org/hwsc-api-blocks/int/hwsc-user-svc/proto"
 	log "github.com/hwsc-org/hwsc-logger/logger"
 	"github.com/hwsc-org/hwsc-user-svc/conf"
-	"github.com/mongodb/mongo-go-driver/mongo"
-	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	// database/sql uses this library indirectly
+	_ "github.com/lib/pq"
 	"os"
 	"os/signal"
 	"syscall"
 )
 
-var (
-	mongoClientReader *mongo.Client
-	mongoClientWriter *mongo.Client
+const (
+	dbDriverName = "postgres"
 )
 
-const (
-	mongoReader = "mongo reader"
-	mongoWriter = "mongo writer"
+var (
+	connectionString    string
+	postgresDB          *sql.DB
+	postgresUnavailable *pb.UserResponse
 )
 
 func init() {
-	log.Info("Connecting to mongo databases")
+	postgresUnavailable = &pb.UserResponse{
+		Status:  &pb.UserResponse_Code{Code: uint32(codes.Unavailable)},
+		Message: codes.Unavailable.String(),
+	}
 
+	log.Info("Connecting to postgres DB")
+
+	// initialize connection string
+	connectionString = fmt.Sprintf(
+		"host=%s user=%s password=%s dbname=%s sslmode=verify-full",
+		conf.UserDB.Host, conf.UserDB.User, conf.UserDB.Password, conf.UserDB.Name)
+
+	// intialize connection object
 	var err error
-	mongoClientReader, err = dialMongoDB(&conf.UserDB.Reader)
+	postgresDB, err = sql.Open(dbDriverName, connectionString)
 	if err != nil {
-		log.Fatal("Failed to connect to mongo reader server:", err.Error())
+		log.Fatal("Failed to intialize connection object:", err.Error())
 	}
 
-	mongoClientWriter, err = dialMongoDB(&conf.UserDB.Writer)
+	// verify connection is alive, establishing connection if necessary
+	err = postgresDB.Ping()
 	if err != nil {
-		log.Fatal("Failed to connect to mongo writer server:", err.Error())
+		log.Fatal("Ping failed, cannot establish connection:", err.Error())
 	}
 
-	// Handle Terminate Signal(Ctrl + C)
+	// Handle Terminate Signal(Ctrl + C) gracefully
 	c := make(chan os.Signal)
 	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
 	go func() {
 		<-c
-		log.Info("Disconnecting mongo databases")
-		_ = disconnectMongoClient(mongoClientReader)
-		_ = disconnectMongoClient(mongoClientWriter)
-		fmt.Println()
+		log.Info("Disconnecting postgres DB")
+		if postgresDB != nil {
+			_ = postgresDB.Close()
+		}
 		log.Fatal("hwsc-user-svc terminated")
 	}()
+
+	// TODO delete after dev-ops working
+	devCreateUserTable()
```

Comment:
Let use Azure Cosmos DB and let the DB persists, rather than creating tables. Please remove this invocation.